syntax = "proto3";

package marketplace;

option go_package = "./marketplace";

// The Marketplace Population Service definition.
service MarketplaceService {
  // PopulateDigitalProducts auto-generates digital products for the marketplace.
  rpc PopulateDigitalProducts (PopulationRequest) returns (PopulationResponse);
}

// Request message for marketplace population.
message PopulationRequest {
  string creator_user_id = 1;
  int32 count = 2;
  string product_category = 3; // e.g., "PLR", "THEMES", "AI_AGENTS"
}

// Response message for marketplace population.
message PopulationResponse {
  bool success = 1;
  string message = 2;
  repeated string product_ids = 3;
}
